### PR TITLE
Add new version string VERSION_DATE_2018_04_05 for NLU

### PR DESCRIPTION
**Pull request by aluu317**
##### Checklist

- [ ] `npm test` passes (tip: `npm run autofix` can correct most style issues)
- [X] tests are included
- [x] documentation is changed or added

##### New version_date Checklist

- [x] A new constant is avaliable with the version_date - [example](https://github.com/watson-developer-cloud/node-sdk/blob/d1418ac2f9774194aaff0c8bd80f0d3722beef72/conversation/v1.js#L77)
- [x] Any older version_date constants remain intact
- [x] The example in the README includes the new version_date constant
- [x] Any relevant code in the examples/ folder has been updated to use the new version_date constant
- [x] Most tests are updated to the new version_date
- [ ] 1-2 new tests are added that use the old version_date (optional, but preferred)

### CODE REVIEW

**Comment by aluu317** [comment]
Can someone help me understand the failure in the Travis build?

**Comment by anweshan** [comment]
Don't worry about the error in the travis build. This is just because the encrypted credentials you are using are not the same encrypted credentials that are in the master branch/the ones referenced in the .travis.yml file

**Comment by germanattanasio** [comment]
@anweshan we should not try to decrypt keys for PRs. The unit tests should run and pass. Can you open an issue and fix the build so that PRs don't fail.

File: natural-language-understanding/v1.ts [hunk 1]
```diff
@@ -20,6 +20,7 @@ import GeneratedNaturalLanguageUnderstandingV1 = require('./v1-generated');
 class NaturalLanguageUnderstandingV1 extends GeneratedNaturalLanguageUnderstandingV1 {
   static VERSION_DATE_2016_01_23: string = '2016-01-23';
   static VERSION_DATE_2017_02_27: string = '2017-02-27';
+  static VERSION_DATE_2018_04_05: string = '2018-04-05';
```
Comment (germanattanasio) [inline natural-language-understanding/v1.ts]: Please don't add the version date as constant. We decided not to use constants anymore to enforce users to type the date.

**Comment by germanattanasio** [comment]
@aluu317 you need to update your git configuration so that your commits are associated with your github account.

See https://help.github.com/articles/setting-your-username-in-git/

File: natural-language-understanding/v1.ts (hunk 1 above)
Comment (aluu317) [reply natural-language-understanding/v1.ts]: Sounds good. I'm confirming if we're removing the constants, does this mean this PR will need to change to pass in the string '2018-04-05' directly instead? https://github.com/watson-developer-cloud/natural-language-understanding-nodejs/pull/23

Comment (anweshan) [reply natural-language-understanding/v1.ts]: @aluu317 Yes! Please change that to just pass in that string.

**Comment by aluu317** [comment]
@germanattanasio @anweshan Can anyone also review this too please? https://github.com/watson-developer-cloud/natural-language-understanding-nodejs/pull/23

**Pull request merged**